packages/monosize-storage-azure/src/uploadReportToRemote.mts

```diff
@@ -14,71 +16,112 @@ export function splitArrayToChunks<T>(arr: T[], size: number): T[][] {
+    const transaction = new TableTransaction();
+    const entitiesIterator = await client.listEntities({
+      queryOptions: {
+        filter: odata`PartitionKey eq ${branch}`,
+      },
+    });
+
+    for await (const entity of entitiesIterator) {
+      // We can't delete and create entries with the same "rowKey" in the same transaction
+      // => we delete only entries not present in existing report
+      const isEntryPresentInExistingReport = Boolean(localReport.find(entry => createRowKey(entry) === entity.rowKey));
+      const shouldEntryBeDeleted = !isEntryPresentInExistingReport;
+
+      if (shouldEntryBeDeleted) {
+        transaction.deleteEntity(entity.partitionKey as string, entity.rowKey as string);
+      }
     }
-  }
 
-  localReport.forEach(entry => {
-    transaction.upsertEntity(
-      {
-        partitionKey: branch,
-        rowKey: createRowKey(entry),
+    localReport.forEach(entry => {
+      transaction.upsertEntity(
+        {
+          partitionKey: branch,
+          rowKey: createRowKey(entry),
 
-        name: entry.name,
-        packageName: entry.packageName,
-        path: entry.path,
+          name: entry.name,
+          packageName: entry.packageName,
+          path: entry.path,
 
-        minifiedSize: entry.minifiedSize,
-        gzippedSize: entry.gzippedSize,
+          minifiedSize: entry.minifiedSize,
+          gzippedSize: entry.gzippedSize,
 
-        commitSHA,
-      },
-      'Replace',
+          commitSHA,
+        },
+        'Replace',
+      );
+    });
+
+    const chunks = splitArrayToChunks(transaction.actions, ENTRIES_PER_CHUNK);
+
+    for (const chunk of chunks) {
+      await client.submitTransaction(chunk);
+    }
+  }
+
+  return uploadReportToRemote;
+}
+
+function getTableClient(authType: AzureAuthenticationType): TableClient {
+  if (typeof process.env['BUNDLESIZE_ACCOUNT_NAME'] !== 'string') {
+    throw new Error('monosize-storage-azure: "BUNDLESIZE_ACCOUNT_NAME" is not defined in your process.env');
+  }
+  const AZURE_STORAGE_ACCOUNT = process.env['BUNDLESIZE_ACCOUNT_NAME'];
+  const AZURE_STORAGE_TABLE_NAME = 'latest';
+
+  if (authType === 'AzurePipelinesCredential') {
+    if (typeof process.env['AZURE_TENANT_ID'] !== 'string') {
+      throw new Error('monosize-storage-azure: "AZURE_TENANT_ID" is not defined in your process.env');
+    }
+    if (typeof process.env['AZURE_CLIENT_ID'] !== 'string') {
+      throw new Error('monosize-storage-azure: "AZURE_CLIENT_ID" is not defined in your process.env');
+    }
+    if (typeof process.env['AZURE_SERVICE_CONNECTION_ID'] !== 'string') {
+      throw new Error('monosize-storage-azure: "AZURE_SERVICE_CONNECTION_ID" is not defined in your process.env');
+    }
+    if (typeof process.env['SYSTEM_ACCESSTOKEN'] !== 'string') {
+      throw new Error('monosize-storage-azure: "SYSTEM_ACCESSTOKEN" is not defined in your process.env');
+    }
```

Comment:
Relevant documentation on where/how to get these values: https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/identity/identity/samples/AzureIdentityExamples.md#authenticating-in-azure-pipelines-with-service-connections

Reply:
lets add those docs into the code as well.

suggestion:

maybe we could fetch all the required env variables and  if one of them would be missing throw error - this could also provide nice DX ouput to user where they would see all the env variables that are missing instead of going through multiple CI failed runs to get things right.

Reply:
created a function to check for **all** required env variables for each auth method first before throwing an error: [489f7f9](https://github.com/microsoft/monosize/pull/79/commits/489f7f9a912b314791245d97c93b47fd683ec2a0)